apps/condo/domains/marketplace/schema/Invoice.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Generated by `createschema marketplace.Invoice 'number:Integer; property:Relationship:Property:PROTECT; unitType:Text; unitName:Text; accountNumber:Text; toPay:Decimal; items:Json; contact?:Relationship:Contact:SET_NULL; client?:Relationship:User:SET_NULL; clientName?:Text; clientPhone?:Text; clientEmail?:Text'`
+ */
+
+const Ajv = require('ajv')
+const addFormats = require('ajv-formats')
+
+const { GQLError, GQLErrorCode: { BAD_USER_INPUT } } = require('@open-condo/keystone/errors')
+const { AutoIncrementInteger } = require('@open-condo/keystone/fields')
+const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
+const { GQLListSchema } = require('@open-condo/keystone/schema')
+
+const { MONEY_AMOUNT_FIELD } = require('@condo/domains/common/schema/fields')
+const { render, getGQLErrorValidator } = require('@condo/domains/common/schema/json.utils')
+const access = require('@condo/domains/marketplace/access/Invoice')
+const {
+    ERROR_INVALID_INVOICE_ROWS,
+    ERROR_NO_INVOICE_RECEIVERS,
+    ERROR_INVOICE_ALREADY_PAID,
+    INVOICE_STATUSES,
+    INVOICE_STATUS_DRAFT, INVOICE_STATUS_PAID,
+} = require('@condo/domains/marketplace/constants')
+
+const ERRORS = {
+    NO_RECEIVERS: {
+        code: BAD_USER_INPUT,
+        type: ERROR_NO_INVOICE_RECEIVERS,
+        message: 'Nobody sees this invoice. You must set property+unitType+unitName or accountNumber',
+        messageForUser: 'api.marketplace.invoice.error.noReceivers',
+    },
+    ALREADY_PAID: {
+        code: BAD_USER_INPUT,
+        type: ERROR_INVOICE_ALREADY_PAID,
+        message: 'Changing of paid invoice is forbidden',
+        messageForUser: 'api.marketplace.invoice.error.alreadyPaid',
+    },
+}
+
+const INVOICE_ROW_GQL_TYPE_NAME = 'InvoiceRowSchemaField'
+const INVOICE_ROW_GQL_INPUT_NAME = 'InvoiceRowSchemaFieldInput'
+
+const invoiceRowSchemaFields = {
+    name: 'String!',
+    toPay: 'String!',
+    count: 'Int!',
+    vat: 'String',
+    salesTax: 'String',
```

Comment:
It’s not required. You can add new row without any sku

Reply:
Test it?

Reply:
Yes. We allow to add rows not from the market catalog. It's ok.
The test for necessary fields was added.